packages/frontend/src/pages/campaigns/index.tsx

```diff
@@ -71,15 +43,38 @@ const STATE_OPTIONS = [
 ];
 
 export const Campaigns = () => {
+    useResetPageScroll();
     const { t } = useTranslation();
-    const { loading, kpiTokens } = useKPITokens();
 
-    const [pageSize, setPageSize] = useState(12);
+    //  fetch KPITokens
+    const [results, setResults] = useState<KPIToken[]>([]);
+    const { searchQuery, setSearchQuery } = useSearch();
+    const { loading, kpiTokens } = useKPITokens(searchQuery);
+
+    //  show filters
     const [filtersOpen, setFilterOpen] = useState(false);
+    const toggleFilters = () => setFilterOpen(!filtersOpen);
+
+    // page settings
+    const [pageSize, setPageSize] = useState(12);
     const [ordering, setOrdering] = useState<SelectOption>(ORDERING_OPTIONS[0]);
     const [state, setState] = useState<SelectOption>(STATE_OPTIONS[0]);
-    const [searchQuery] = useState("");
-    const [, /* debouncedSearchQuery */ setDebouncedSearchQuery] = useState("");
+
+    // filter and sort results
+    const filteredTokens = useMemo(() => {
+        return filterKPITokens(Object.values(kpiTokens), undefined);
```

Comment:
Should we always pass undefined to filterKPITokens?

Reply:
Don't think so. I'll remove. thanks.

Reply:
Fixed.